### PR TITLE
Testing feature

### PR DESCRIPTION
**Pull request by elberthcorniell**
In this pull request we want to add testing features to this Enumerable methods:

- Each;
- Each with index;
- Select;
- All;
- Any;
- None;
- Count;
- Map;
- Inject.

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    expectations.include_chain_clauses_in_custom_matcher_descriptions = true
+  end
+  config.mock_with :rspec do |mocks|
+    mocks.verify_partial_doubles = true
+  end
+  config.shared_context_metadata_behavior = :apply_to_host_groups
+end
+
+require_relative '../enumerables.rb'
+describe "Enumerablea" do
```
Comment (kitpao) [inline spec/spec_helper.rb]: - [x] **OPTIONAL** You may want to correct "Enumerablea" to "Enumerables"

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+      expect((1..5).my_inject { |sum, n| sum + n }).to eql 15
+    end
+    it "Works when a block and an Argument are given" do
+      expect((1..5).my_inject(1) { |product, n| product * n }).to eql 120
+    end
+    it "Works when a a symbol is given and no block is given" do
+      expect((1..5).my_inject(1, :*)).to eql 120
+    end
+  end
+end
```
Comment (kitpao) [inline spec/spec_helper.rb]: - [x] Please, do not use the spec_helper file to add your tests, your tests should be in another file inside the spec directory (for example enumerables_spec.rb). This is for convention and file organization (spec_helper is for running the configurations, everything else ending with '_spec' is for the tests). Kindly move your tests to a new file.

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+      i = 0
+      arr.my_each { |n| expect(n).to eql(arr[i]); i += 1 }
+    end
+    it "Goes through all elements in an Hash" do
+      hash.my_each { |n| expect(n[1]).to eql(hash[n[0]])}
+    end
+    it "Goes through all elements in an Range" do
+      i = 0
+      range.my_each { |n| expect(n).to eql(i); i += 1 }
+    end
```
Comment (kitpao) [inline spec/spec_helper.rb]: - [x] This test doesn't prove that your method works with ranges, if I put `let!(:range) { 2..5 }` It is not working. Maybe you want to use in your test `eql(range.to_a[i])` so it adapts to the given range, same logic applies to my_each_with_index

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+    it "Returns an Hash when self is an Hash and a blog is given" do
+      result = hash.my_each { |n| n }
+      expect(result.kind_of?(Hash)).to eql(true)
+    end
+    it "Returns an Range when self is an Range and a blog is given" do
+      result = range.my_each { |n| n }
+      expect(result.kind_of?(Range)).to eql(true)
+    end
+  end
```
Comment (kitpao) [inline spec/spec_helper.rb]: - [x] **OPTIONAL** You may want to correct "blog" to "block"

**Review by kitpao — CHANGES_REQUESTED** [review]
# 👋 Hello @elberthcorniell and @anapdh 

I see you made a great effort into putting these tests to work 👏 There are some technical conventions that you are missing to implement among other details, please check them below 🔽 

**_Note:_** All the **[OPTIONAL]** are **not required** in order to approve your project but are personal suggestions

- [x] Do not forget to add the Gemfile, this file is where all your dependencies will be stored so other people can use your repo with no issues. (In this case, to add the RSpec gem for testing) You can learn more [here](https://bundler.io/gemfile.html). Please add the Gemfile with the necessary gem :)

- [x] Now that you have your tests, please add the corresponding instructions in your README file, remember you can put code inside backquotes to format it `like this`. The instructions would be to run bundle install and then type rspec in terminal

- [x] Please include negative scenarios in your tests (what happens when invalid inputs are given? or what answers are wrong?) You can use more variety in the RSpec commands to achieve this, [here you can find a cheat sheet to see all the options you have for testing](https://kapeli.com/cheat_sheets/RSpec_Expectations.docset/Contents/Resources/Documents/index)
image
  > Example: there is a spec for correct usage of the method [e.g. it "creates object if all required attributes are passed"]
and also there is spec for the wrong usage of a method [e.g. it "does not create the object if required attributes are missing"].

- [x] Now that you have your automated tests, you may delete your test.rb file, or add it to the gitignore (I see your test.rb file has much more testing scenarios if you want you can add them to your rspec tests so you won't loose them)
image

- [x] You forgot to add your linters.yml file in your project, please add it, [you can find it here](https://github.com/microverseinc/linters-config/blob/master/ruby/.github/workflows/linters.yml). And kindly fix the indicated offenses

- [x] You are missing the tests for the multiply_els method

- Do not forget the inline comments below ⬇️ 

### ♻️ Status: Your project requires some additional changes ♻️
➡️ Please request another review once you are ready 📝 
___
##### 👉 If you have any doubts you can contact me on slack as @kitpao
##### 🌟  If you have feedback about my reviews I would appreciate it if you could leave a recommendation in LinkedIn 💌 or let me know in Slack if you see improvements to be done 😄  

🔗 __[Github](https://github.com/kitpao) | [Twitter](https://twitter.com/Kitpao1) | [LinkedIn](https://www.linkedin.com/in/kitzia-paola-vidal/) | [Medium](https://medium.com/@kitpao91)__

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello Imran Yousaf & Elbert Corniell Caceres** 
Good job with the improvements to your project :muscle:  :muscle:
There are some issues to correct before you can go ahead:

- [x] Fix all rubocop offenses
- [ ] For each method, cover both positive and negative scenarios, you might use "to_not eql" or "to_not be_an".
- [x] Don't use multiple "expect" elements inside an "it" element.
- [x] In the README file, add a screenshot.
 
Please request another review once you are ready! :+1: 
 
Happy coding :relaxed:
 
Oscar Andrés Russi Porras

File: .rubocop.yml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+Metrics/BlockLength:
+  Enabled: false
+
+Lint/BinaryOperatorWithIdenticalOperands:
+  Enabled: false
```
Comment (SpaYco) [inline .rubocop.yml]: - [x] 🛑 don't edit rubocop configs, please get it back to what's provided by Microverse [here](https://github.com/microverseinc/linters-config/tree/master/ruby) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/linter_config.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/))
_we use a tool to check this, so please don't try to sneak a line in there_

File: Gemfile [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+source 'https://rubygems.org'
+
+git_source(:github) { |repo_name| "https://github.com/#{repo_name}" }
+
+gem 'rspec'
+gem 'rubocop', require: false
+
+# gem "rails"
```
Comment (SpaYco) [inline Gemfile]: - [x] delete unused code. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

**Review by SpaYco — CHANGES_REQUESTED** [review]
# Changes Requested 

Hello Team! 👋 
Awesome job so far!! 👍 you still need to make some changes though! 

## Required Changes ♻️ 

- [x] check below 👇 

### please make the appropriate changes and ask for another code review ✔️ 

Happy Coding!

Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

File: .rubocop.yml (hunk 5 above)
Comment (elberthcorniell) [reply .rubocop.yml]: I didn't try to sneak anything. We just didn't know microverse provided a config file.

Comment (SpaYco) [reply .rubocop.yml]: @elberthcorniell sorry if that came out wrong, I didn't accuse you of that, I'm just letting you know

**Review by SpaYco — APPROVED** [review]
# Project Approved ✔️ 

thank you for making the required changes, your project is now approved!

### please don't forget to merge to master, good luck with the next projects!


Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Pull request merged**